roles/manage-gcp-infra/tasks/configure-registry-object-storage.yaml

```diff
@@ -0,0 +1,67 @@
+---
+- name: retrieve the name of one registry pod
+  shell: oc get pod -l deploymentconfig=docker-registry | grep -m 1 registry | awk '{print $1}'
+  register: pod  
+  
+- name: create a temp directory fror the google templates
+  file:
+    path: /tmp/gcp_templates
+    state: directory
+    mode: 0755  
+  
+- name: retrieve the configuration file
+  shell: oc rsync {{ pod.stdout }}:/config.yml /tmp/gcp_templates -n default
+    
+- name: replace with google object storage configuration 1
+#  shell: "sed -i {{ gcloud_sed_expression1 }} /tmp/gcp_templates/config.yml && sed -i {{ gcloud_sed_expression2 }} /tmp/gcp_templates/config.yml"  
+  lineinfile:
+    path: /tmp/gcp_templates/config.yml
+    regexp: "filesystem"
+    state: present
+    line: "  gcs:" 
+
+- name: replace with google object storage configuration 2
+#  shell: "sed -i {{ gcloud_sed_expression1 }} /tmp/gcp_templates/config.yml && sed -i {{ gcloud_sed_expression2 }} /tmp/gcp_templates/config.yml"  
+  lineinfile:
+    path: /tmp/gcp_templates/config.yml
+    regexp: "rootdirectory"
+    state: present
+    line: "    bucket: openshift-{{ gcloud_env }}-registry"           
+  
+- name: check if configmap exists
+  shell: oc get configmap registry-config -n default | grep registry-config  
+  register: configmap
+  changed_when: false
+  failed_when: not (configmap.rc == 0 or configmap.rc == 1)
+  
+- name: delete configmap if it exists
+  shell: oc delete configmap registry-config -n default
+  when: configmap.stdout  
+  
+- name: create configmap with new config
+  shell: oc create configmap registry-config --from-file=/tmp/gcp_templates/config.yml -n default
+  
+- name: check if current registry storage exists  
+  shell: oc set volume dc/docker-registry -n default | grep registry-storage
+  register: registry_storage
+  changed_when: false
+  failed_when: not (registry_storage.rc == 0 or registry_storage.rc == 1)
+  
+- name: remove current registry storage if present
+  shell: oc set volume dc/docker-registry --remove --name=registry-storage -n default
+  when: registry_storage.stdout
+  
+- name: check if configmap currently mounted
+  shell: oc set volume dc/docker-registry -n default | grep "/etc/docker/registry"
```

Comment:
same as above -can we please look to *not* use the `set` command to perform the check ?

Reply:
set used this way can be used to list volumes and it does not have side effects.

Reply:
see above